sphinx/ext/autosummary/generate.py

```diff
@@ -221,12 +217,19 @@ def find_autosummary_in_files(filenames):
     """
     documented = []
     for filename in filenames:
-        f = open(filename, 'r')
-        lines = f.read().splitlines()
-        documented.extend(find_autosummary_in_lines(lines, filename=filename))
-        f.close()
+        with open(filename, 'rb') as f:
+            lines = f.read()
+            try:
+                # sometimes this fails (e.g., Windows Py3k w/utf-8 files)
+                lines = lines.decode()
+            except Exception:
+                lines = lines.decode('utf-8')
```

Comment:
@birkenfeld is this better? I think this version should be Python 2 compatible, as well. I can test on a Python 2 system tomorrow.

Now the decoding first tries to proceed the way it would have before (via an implicit `.decode()`), and if it fails, tries a different decode method (`utf-8`). I think this should be at least as compatible as what was there before. Let me know if you have a better idea.

Do you want me to `try/except the `lines = lines.decode('utf-8')` as well, and throw an informative error if the decoding fails? If so, let me know what you'd suggest for the error line (type and text) and I'll put it in.